plugins/repository.go

```diff
@@ -0,0 +1,101 @@
+package plugins
+
+import (
+	"errors"
+	"github.com/spf13/cobra"
+	"log"
+	"os"
+	"path/filepath"
+	"sync"
+)
+
+const argRepository = "path"
+
+type RepositoryPlugin struct {
+	Plugin
+	Path string
+}
+
+func (p *RepositoryPlugin) IsEnabled() bool {
+	return p.Enabled
+}
+
+func (p *RepositoryPlugin) DefineCommandLineArgs(cmd *cobra.Command) error {
+	flags := cmd.Flags()
+	flags.String(argRepository, "", "scan repository folder")
+	return nil
+}
+
+func (p *RepositoryPlugin) Initialize(cmd *cobra.Command) error {
+	flags := cmd.Flags()
+	directoryPath, _ := flags.GetString(argRepository)
+	if directoryPath == "" {
+		return errors.New("path to repository missing. Plugin initialization failed")
+	}
+
+	p.Path = directoryPath
+	p.Enabled = true
+	return nil
+}
+
+func (p *RepositoryPlugin) GetItems(items chan Item, errs chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	wg.Add(1)
+	go p.getFiles(items, errs, wg)
+}
+
+func (p *RepositoryPlugin) getFiles(items chan Item, errs chan error, wg *sync.WaitGroup) {
+	defer wg.Done()
+	fileList := make([]string, 0)
+	err := filepath.Walk(p.Path, func(path string, fInfo os.FileInfo, err error) error {
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+		if fInfo.Name() == ".git" && fInfo.IsDir() {
+			return filepath.SkipDir
+		}
+		if fInfo.Size() == 0 {
+			return nil
+		}
+		if !fInfo.IsDir() {
+			fileList = append(fileList, path)
+		}
+		return err
+	})
+
+	if err != nil {
+		panic(err)
```

Comment:
You should use `zerolog` for logging, and you can use `log.Fatal` to log and exit the program.

Reply:
was making some tests when I made this and the panic got trough. Thanks for noticing it!

Reply:
But you still using `log`, and it is not the `zerolog` logger, right?
Or I'm missing something here?

Reply:
this is the outdated version, for some reason it did not update, but check the files section to see the change: https://github.com/Checkmarx/2ms/pull/62/files

Reply:
See my commit added to this PR: https://github.com/Checkmarx/2ms/pull/62/commits/98cb6ef80de78ee8c6002e8cfdde6d53f0270104